src/Compilers/Core/Portable/DiagnosticAnalyzer/AsyncQueue.cs

```diff
@@ -140,6 +146,11 @@ public void Complete()
             }
         }
 
+        public void PromiseNotToEnqueue()
```

Comment:
That is possible, but it would require a more invasive reworking of the API, and I'd prefer not to do that on the stabilization branch.